python/activator/middleware_interface.py

```diff
@@ -1262,11 +1262,25 @@ def _check_permanent_changes(self, where: str) -> bool:
         changes : `bool`
             `True` if changes have been made, `False` if retries are safe.
         """
-        data_ids = self.butler.query_data_ids(["instrument", "visit", "detector"], where=where, explain=False)
+        # Need dimension records to determine region.
+        data_ids = self.butler.query_data_ids(
+            ["instrument", "visit", "detector"], where=where, with_dimension_records=True, explain=False
+        )
         if len(data_ids) == 1:
             data_id = data_ids[0]
+            # Check if processing happened already, this needs visit timestamp.
+            # Use begin or end of timespan, whichever is defined.
+            visit_time: astropy.time.Time | None = None
+            if data_id.timespan is not None:
+                visit_time = data_id.timespan.begin or data_id.timespan.end
+            if visit_time is None:
+                # Without timespan pipelines will not be able to run, so just
+                # stop here.
+                raise ValueError(f"No timespan defined for this visit: {data_id}")
```

Comment:
This comment is incorrect; this method has nothing to do with running pipelines. I suggest simply deleting it.

In any case, it's problematic to raise an exception from error handling code. The best response (same as in the case where _multiple_ data IDs are found) is to log a warning and return `True`.

Reply:
OK, will change to warning and return True.